libmozevent/utils.py

```diff
@@ -18,15 +19,22 @@
 
 def run_tasks(awaitables: Iterable):
     """
-    Helper to run tasks concurrently, but when an exception is raised
-    by one of the tasks, the whole stack stops.
+    Helper to run tasks concurrently
+    When an exception is raised by one of the tasks or a SIGTERM signal is received,
+    the whole stack stops and an asyncio.CancelledError exception is raised
     """
+    event_loop = asyncio.get_event_loop()
+
+    # Create a task grouping all awaitables and running them concurrently
+    task = asyncio.gather(*awaitables)
+
+    def handle_sigterm(*args, **kwargs):
+        task.cancel()
```

Comment:
I would add some log there, at least at INFO level to know that we got a sigterm. That's one of the issues currently, we do not have early info about the SIGTERM